### PR TITLE
Rebase to v3.4.9

### PR DESCRIPTION
**Pull request by dscho**
This requires https://github.com/msys2/msys2-runtime/pull/172 to be merged first, and, depending in which way it was "merged" (whether by pushing directly to `msys2-3.4.9` or by hitting the "Rebase and merge" button), I will need to re-rebase to include the actual `msys2-3.4.9` branch.

<details><summary>Range-diff relative to `main`</summary>

* 1:  52012877a0 =  1:  01f7ce4fd1 Add MSYS2 triplet
* 2:  e84fccba60 =  2:  04e699b587 Fix msys library name in import libraries
* 3:  e7ff93755b =  3:  bbedcadbf7 Rename dll from cygwin to msys
* 4:  42631a088d !  4:  6fe093f16d Add functionality for converting UNIX paths in arguments and environment variables to Windows form for native Win32 applications.

   ``````diff
   @@ winsup/cygwin/msys2_path_conv.cc (new)
    +
    +
    +void find_end_of_posix_list(const char** to, int* in_string) {
   -+    for (; **to != '\0' && (in_string ? (**to != *in_string) : **to != ' '); ++*to) {
   ++    for (; **to != '\0' && (!in_string || **to != *in_string); ++*to) {
    +    }
    +
    +    if (**to == *in_string) {
   @@ winsup/cygwin/msys2_path_conv.cc (new)
    +            }
    +            continue;
    +        }
   -+
   -+        if (isspace(*srcit)) {
   -+            //sub_convert(&srcbeg, &srcit, &dstit, dstend, &in_string);
   -+            //srcbeg = srcit + 1;
   -+            break;
   -+        }
    +    }
    +
    +    sub_convert(&srcbeg, &srcit, &dstit, dstend, &in_string);
   ``````

* 5:  f128c3d95b =  5:  2a23aebb70 Add functionality for changing OS name via MSYSTEM environment variables.
* 6:  8b6eefea34 =  6:  d7d7c59cc5 - Move root to /usr. - Change sorting mount points. - By default mount without ACLs. - Can read /etc/fstab with short mount point format.
* 7:  0fc4cf316c =  7:  6085c158cc Instead of creating Cygwin symlinks, use deep copy by default
* 8:  0d77252ed7 =  8:  b2f377fd24 Automatically rewrite TERM=msys to TERM=cygwin
* 9:  45aa3867bb =  9:  3ddd8cd712 Do not convert environment for strace
* 10:  d9d7bda19a = 10:  2b62456ea7 path_conv: special-case root directory to have trailing slash
* 11:  8a7e2c8fdc = 11:  79a9de58c4 dcrt0.cc: Untangle allow_glob from winshell
* 12:  48bf4db48a = 12:  549cc7edeb dcrt0.cc (globify): Don't quote literal strings differently when dos_spec
* 13:  d3686a1495 = 13:  8709ebbcfc strace.cc: Don't set MSYS=noglob
* 14:  fdf8171fe8 = 14:  04cf05a93f Add debugging for build_argv
* 15:  5d8a5b29aa = 15:  baace0b995 Add debugging for strace make_command_line
* 16:  6d550baa78 = 16:  b672399927 environ.cc: New facility/environment variable MSYS2_ENV_CONV_EXCL
* 17:  8573928f8e = 17:  12198197ec Fix native symbolic link spawn passing wrong arg0
* 18:  b81a4da46a = 18:  4aa083ec34 strace --quiet: be *really* quiet
* 19:  cb4a389838 = 19:  53cfd18bd9 Introduce the `enable_pcon` value for `MSYS`
* 20:  d0804ef6bc = 20:  e70cae27a5 popen: call /usr/bin/sh instead of /bin/sh
* 21:  c306256838 = 21:  c5e1e1373e Disable the 'cygwin' GitHub workflow
* 22:  a9ff81717d ! 22:  c9299efebc CI: add a GHA for doing a basic build test

   ``````diff
   @@ Metadata
     ## Commit message ##
        CI: add a GHA for doing a basic build test
    
   +    Build with --disable-dependency-tracking because we only build once
   +    and this saves 3-4 minutes in CI.
   +
     ## .github/workflows/build.yaml (new) ##
    @@
    +name: build
   @@ .github/workflows/build.yaml (new)
    +        shell: msys2 {0}
    +        run: |
    +          (cd winsup && ./autogen.sh)
   -+          ./configure
   ++          ./configure --disable-dependency-tracking
    +          make -j8
    +
    +      - name: Install
   ``````

* 23:  a94b29fa4f = 23:  e1a8db5dc7 Set up a GitHub Action to keep in sync with Cygwin
* 24:  38482c8a8a = 24:  50478bbe56 Expose full command-lines to other Win32 processes by default
* 25:  f7fc589306 = 25:  a438c32ca5 Do not show Error dialogs by default
* 26:  502a5cb189 = 26:  771bea432f Add a helper to obtain a function's address in kernel32.dll
* 27:  c2baa124a2 = 27:  2befb21ebb Emulate GenerateConsoleCtrlEvent() upon Ctrl+C
* 28:  c7d368c6f9 = 28:  5e85ca5b44 kill: kill Win32 processes more gently
* 29:  d5fafdee6b = 29:  878160a8d6 Cygwin: make option for native inner link handling.
* 30:  c161d364a3 = 30:  9fbbdf6061 docs: skip building texinfo and PDF files
* 31:  5b68740a7b = 31:  82bbdeaf6b install-libs: depend on the "toollibs"
* 32:  d650381271 = 32:  a7e656a671 POSIX-ify the SHELL variable
* 33:  92f22cf074 = 33:  3c2b1f3fab Handle ORIGINAL_PATH just like PATH
* 34:  71a9122b45 = 34:  723401047a uname: allow setting the system name to CYGWIN
* 35:  10a1ce2b18 = 35:  102d9a2ee6 Pass environment variables with empty values
* 36:  0a4800c492 = 36:  0924ea36bc Optionally disallow empty environment values again
* 37:  c3f0426e77 = 37:  e4cf39bf0b build_env(): respect the `MSYS` environment variable
* 38:  1520489bd4 = 38:  5db940f08a When converting to a Unix path, avoid double trailing slashes
* 39:  0ec2f450dc = 39:  eb19539016 msys2_path_conv: pass PC_NOFULL to path_conv
* 40:  a556be71db = 40:  d325910f42 Revert "Cygwin: Enable dynamicbase on the Cygwin DLL by default"
* 41:  2a34d37c67 <  -:  ---------- CI: build with --disable-dependency-tracking
* 42:  2078f6b88f <  -:  ---------- Stop assuming that there are no spaces in POSIX-style paths
* -:  ---------- > 41:  d52da7859b dumper: avoid linker problem when `libbfd` depends on `libsframe`
* 43:  5929cd074e = 42:  6b376b9f5b Handle 8-bit characters under LOCALE=C
* 44:  fde6d29aa3 = 43:  3e71002972 Mention the extremely useful small_printf() function
* 45:  c02f0cda99 = 44:  63c0e97734 Leave arguments starting with a tilde or quote alone
* 46:  b634c462fc = 45:  f90152f456 Leave Git's :name and :/message arguments alone, please
* 47:  76393d2d70 = 46:  e03e374567 Leave paths containing any special characters alone
* 48:  28817d96dc = 47:  0ea33f2531 Leave paths containing '/~' alone
* 49:  bf69c7dc3a = 48:  eb0accdef0 Skip posix-to-windows conversion when '::' is seen
* 50:  37040d5494 = 49:  d8a47ae941 Also leave Git's <rev>:./<name> syntax alone
* 51:  b851a9fe14 = 50:  6ee09769cf Arguments starting with '@@' are no paths
* 52:  6ecfeb8e80 = 51:  39910e3b35 Prevent scp-style arguments from being mangled
* 53:  2d48392b41 = 52:  69472494a4 Allow overriding the home directory via the HOME variable
* 54:  bca5d76c2b = 53:  9fb1991862 Fixed path converting with non ascii char.
* 55:  c064185bd2 = 54:  b70eab50db Respect `db_home` setting even for the SYSTEM account
* 56:  945e273da3 = 55:  49602d166c path-conversion: Introduce ability to switch off conversion.
* 59:  9dd92bf8d5 = 56:  0807f4b733 Respect the `db_home: env` setting under more circumstances
* 60:  7272586f58 = 57:  a92e510082 Allow native symlinks to non-existing targets in 'nativestrict' mode
* 57:  370429742f = 58:  0362055c9f Make paths' WCS->MBS conversion explicit
* 58:  0c86160814 = 59:  4d86367cd4 Avoid sharing cygheaps across Cygwin versions
* 61:  a7935c5df5 = 60:  9ad0baec21 Use MB_CUR_MAX == 6 by default
* 62:  21494118f1 = 61:  d8f626b217 Change the default base address for x86_64
* 63:  3d9fa6255f = 62:  4dcf1bddfe Do not try to sync with Cygwin
* 64:  1d9a89814e ! 63:  dbd0a16e27 uname: report msys2-runtime commit hash, too

   ``````diff
   @@ winsup/cygwin/scripts/mkvers.sh: then
    
     ## winsup/cygwin/uname.cc ##
    @@ winsup/cygwin/uname.cc: uname_x (struct utsname *name)
   - 	snprintf (name->release, _UTSNAME_LENGTH, "%s.%s",
   - 		  uname_dev_version, name->machine);
   + #pragma GCC diagnostic push
   + #pragma GCC diagnostic ignored "-Wformat-truncation="
   + #ifdef CYGPORT_RELEASE_INFO
   +-      snprintf (name->release, _UTSNAME_LENGTH, "%s.%s",
   +-		__XSTRING (CYGPORT_RELEASE_INFO), name->machine);
   ++      snprintf (name->release, _UTSNAME_LENGTH, "%s-%s.%s",
   ++		__XSTRING (CYGPORT_RELEASE_INFO), MSYS2_RUNTIME_COMMIT_SHORT, name->machine);
   + #else
   +       extern const char *uname_dev_version;
   +       if (uname_dev_version && uname_dev_version[0])
   +-	snprintf (name->release, _UTSNAME_LENGTH, "%s.%s",
   +-		  uname_dev_version, name->machine);
   ++	snprintf (name->release, _UTSNAME_LENGTH, "%s-%s.%s",
   ++		  uname_dev_version, MSYS2_RUNTIME_COMMIT_SHORT, name->machine);
           else
    -	__small_sprintf (name->release, "%d.%d.%d-api-%d.%s",
    +	__small_sprintf (name->release, "%d.%d.%d-%s-api-%d.%s",
   ``````

* 65:  77ef46a182 = 64:  10d77b53da Revert "Cygwin: pipe: Restore blocking mode for cygwin process at startup."
* 66:  aa056f2045 <  -:  ---------- fchmodat/fstatat: fix regression with empty `pathname`
* 67:  0faeaafc21 <  -:  ---------- fixup! uname: report msys2-runtime commit hash, too
* 68:  76b408b5d4 <  -:  ---------- fixup! fchmodat/fstatat: fix regression with empty `pathname`
* 69:  6c91faa4b8 <  -:  ---------- Cygwin: gen_full_path_at: drop never reached code
* 70:  c606f1dd38 <  -:  ---------- Define _AT_NULL_PATHNAME_ALLOWED
* 71:  cb6bc405bb <  -:  ---------- Cygwin: use new _AT_NULL_PATHNAME_ALLOWED flag
* 72:  047b54131e <  -:  ---------- Cygwin: Fix and streamline AT_EMPTY_PATH handling
* 73:  eb7ffd4130 ! 65:  a9a875c653 msys2-runtime: restore fast path for current user primary group

   ``````diff
   @@ winsup/doc/ntsec.xml: The two lines starting with the keywords <literal>passwd:<
    -for local accounts, Active Directory for domain account.  Examples:
    +for local accounts, Active Directory for domain account.  For the current
    +user, the default group is obtained from the current user token to avoid
   -+additional lookups to the group database. <literal>db-accuarte</literal> 
   ++additional lookups to the group database. <literal>db-accurate</literal>
    +is only valid on <literal>group:</literal> line, and performs the same 
    +lookups as the <literal>db</literal> option, but disables using the
    +current user token to retrieve the default group as this optimization
   ``````

* 74:  e2a1cb4580 <  -:  ---------- fixup! msys2-runtime: restore fast path for current user primary group

</details>

tl;dr the `_AT_EMPTY_PATH` stuff is now in Cygwin proper and no longer necessary (hence looks like it was dropped), and a couple of `fixup!`s were squashed (both in MSYS2 and in Git for Windows' parts of the branch thicket).

### CODE REVIEW

**Comment by dscho** [comment]
Nice: The branch thicket on top of MSYS2 looks much more put together now:

git-for-windows-branch-thicket
